src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -66,6 +66,22 @@ function CartesianPoint(x::TX, y::TY, z::TZ) where {TX<:Real,TY<:Real,TZ<:Real}
     CartesianPoint{T}(T(x),T(y),T(z))
 end
 
+#Unit support
+function CartesianPoint(x::Unitful.Quantity, y::Unitful.Quantity, z::Unitful.Quantity)
+    for (i, pt) in enumerate((x, y, z))
+        !(pt isa Unitful.Length) && throw(ArgumentError(
+            "Coordinate $i must be a length, got $(pt) with unit $(Unitful.unit(pt))"
+        ))
+    end
+    CartesianPoint(x::Unitful.Length, y::Unitful.Length, z::Unitful.Length)
+end
+
+function CartesianPoint(x::Unitful.Length, y::Unitful.Length, z::Unitful.Length)
+    vals = to_internal_units.( (x, y, z) )
+    T = all(v -> v isa Float32, vals) ? Float32 : Float64
+    CartesianPoint{T}(T.(vals)...)
+end
```

Comment:
With this implementation, I'm getting unexpected type results:
```julia
CartesianPoint(1u"m", 2u"m", 3f0u"m") # CartesianPoint{Float64}(1.0, 2.0, 3.0)
```
is of type `Float64`, even though I only passed `Ints` and `Float32`.

We should add this case to the tests.